pkg/daemon/server/service/healthStatus.go

```diff
@@ -0,0 +1,456 @@
+	Status string `json:"status"`
+	// Message is the error message if any
+	Message string `json:"message"`
+	// Code is the status code for the data health
+	Code string `json:"code"`
+}
+
+// NewDataHealthResponse is used to create a new DataHealthResponse object
+func NewDataHealthResponse(status string, message string, code string) *DataHealthResponse {
+	return &DataHealthResponse{
+		Status:  status,
+		Message: message,
+		Code:    code,
+	}
+}
+
+// DefaultDataHealthResponse is the default response returned by the data health check API
+var DefaultDataHealthResponse = NewDataHealthResponse(PipelineStatusOK,
+	"Pipeline dataflow is healthy",
+	"D1")
+
+// HealthChecker is the struct type for health checker.
+type HealthChecker struct {
+	// Add a field for the health status.
+	currentDataStatus *DataHealthResponse
+	isbSvcClient      isbsvc.ISBService
+	pipeline          *v1alpha1.Pipeline
+	timelineData      map[string]*sharedqueue.OverflowQueue[*TimelineEntry]
+	statusLock        *sync.RWMutex
+}
+
+// NewHealthChecker creates a new object HealthChecker struct type.
+func NewHealthChecker(pipeline *v1alpha1.Pipeline, isbSvcClient isbsvc.ISBService) *HealthChecker {
+	// Return a new HealthChecker struct instance.
+	return &HealthChecker{
+		currentDataStatus: DefaultDataHealthResponse,
+		isbSvcClient:      isbSvcClient,
+		pipeline:          pipeline,
+		timelineData:      make(map[string]*sharedqueue.OverflowQueue[*TimelineEntry]),
+		statusLock:        &sync.RWMutex{},
+	}
+}
+
+// VertexState is a struct which contains the name and  state of a vertex
+type VertexState struct {
+	// Name is the name of the vertex
+	Name string `json:"name"`
+	// State is the state of the vertex
+	State string `json:"state"`
+}
+
+// NewVertexState is used to create a new VertexState object
+func NewVertexState(name string, state string) *VertexState {
+	return &VertexState{
+		Name:  name,
+		State: state,
+	}
+}
+
+// GetCurrentHealth returns the current health status of the pipeline.
+// It is thread safe to ensure concurrent access.
+func (hc *HealthChecker) GetCurrentHealth() *DataHealthResponse {
+	// Lock the statusLock to ensure thread safety.
+	hc.statusLock.RLock()
+	defer hc.statusLock.RUnlock()
+	// Return the health status.
+	return hc.currentDataStatus
+}
+
+// SetCurrentHealth sets the current health status of the pipeline.
+// It is thread safe to ensure concurrent access.
+func (hc *HealthChecker) SetCurrentHealth(status *DataHealthResponse) {
+	// Lock the statusLock to ensure thread safety.
+	hc.statusLock.Lock()
+	defer hc.statusLock.Unlock()
+	// Set the health status.
+	hc.currentDataStatus = status
+}
+
+// StartHealthCheck starts the health check for the pipeline.
+// The ticks are generated at the interval of HealthTimeStep.
+func (hc *HealthChecker) StartHealthCheck(ctx context.Context) {
+	// Goroutine to listen for ticks
+	// At every tick, check and update the health status of the pipeline.
+	go func(ctx context.Context) {
```

Comment:
Shouldn't the function `StartHealthCheck` be blocked? It's using a goroutine to start it in the daemon server.

Reply:
Updated. Please check if it seems fine now